### ISSUE TITLE
Split previous issues list into separate issues

### ISSUE THREAD

**Issue by davidryan59** [davidryan59/ji-ear-trainer#1]
**Currently notes are in the README!**

Currently implementing a Checkbox system
in parallel with the Picklist system
...

Presets

Currently we are using preset 'Custom'
Have more presets:
- Interval in Octave
- Triad
- Four note splodge
- etc

Buttons for:
- Utonal display
- Compound fraction display
- Cents display
- Stats display

### Prime include / exclude

Do some 3-limit 5-limit options
NO LIMIT
prime limits: 3, 5, 7, 11, 13
primesToInclude
primesToExclude
(Probably 1 dropdown to control both options)

Also consider doing odd limits:
- Odd limit 9, 15, 25, 27, 81
- Have a think about whether this really adds anything to the program?

Chord calculation algorithm - make 1 more option, a time constraint, defaulting to say 200ms
- Not sure this is needed? 500000 ticks already sort of does it

Make it possible to play all intervals after question has been answered (rightly/wrongly)

Far fewer components disappearing. Use CSS transitions to animate smoothly. Buttons could change into each other.

Play button
- It should be possible to end playback early
- This also ought to stop the sound
- Move sound stopping to a separate thunk?
- Also, shouldn't be able to change mode until playback has ended (otherwise action kicks in on a new screen, where its no longer needed)

Utonal note - top border of ratio wrong after question answered

Might want to have a picklist with various otonality / utonality ranges
(eg. ALL, Otonal only, Utonal only)

Change name of this project to 
Chord Trainer
Chord and Interval Trainer
Chord and Interval Trainer in Just Intonation
CITJI
CTJI

## Test setup Parameters
- Want a calculate button to calculate all possible chords
- Want a start test button, which fixes the set of chords and tests you on them.
  
## Controls during test
- Minimum frequency (100 Hz)
- Maximum frequency (1000 Hz)
  

For 3+ notes, play the notes in random order
- Write a function that sorts [[0, rand], [1, rand]... [n-1, rand]] by the rands, then extracts the first elements. This is a permutation function.
- Use that to access random indices in the array

Selection
- Might want to add a few more in, e.g. 2:3 ought to be next to 27:40 (which has high complexity)

### Animation
Controlled by the transition line and margin lines below. Probably need a classname on / off to control the transition.
.wrapper {
    display: flex;
    width: 100%;
    height: 100%;
    transition: margin 2s;
    margin: 0 0 0 -250px;
}
.wrapper.is-nav-open {
    margin-left: 0;
}

Keyboard
- PLay next one with P
- Select answers 1 to 6 with numeric keys

ON Mark, want a 3 step flow, can be controlled via chain of 3 dispatches:
- Question is marked in place. All question colours appear
- Tick buttons disappear
- Section moves down (animation if possible)
- And then new row appears

Want a difficulty measure on there - what is the minimum number of cents difference from the correct answer?
- < version for minimum difficulty across chord (before reveal)
- = version for actual difficulty of the correct value from its nearest neighbour

Dropdown to control how many notes there are

Want to be able to play each option
- This should be enabled in practice mode, and disabled in test mode
- Dropdown for practice / test mode?

## Generation of intervals / chords
- Might be better to use all the intervals from 1 to 10, and then all the ratios between them, than the Stern Brocot tree?
- Want utonal to appear as often as otonal
- Can what's done with 2 be generalised to chords of 3+?

Proper interval generation will fix this BUG: currently it is possible for two chords to be the same (e.g. 4:5:6, 4:5:6)
Need to make a list of all distinct chords, and then choose subsets.
- (This is fixed for 2 notes - Stern Brocot)

Go through 'removed' directories. Remove anything I don't think will get used.
